tests-beta/android/MASVS-CODE/MASTG-TEST-0290.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Enforced Application Updates
+platform: android
+id: MASTG-TEST-0290
+type: [static]
+weakness: MASWE-0036
+profiles: [L1, L2]
+---
+
+## Overview
+
+Enforced updates on Android occur when an application requires users to install the latest version before they can continue using it. This behavior is commonly implemented via the Google Play Core In-App Update API using the Immediate update type with a value `1`. When the Immediate update flow is triggered, the application blocks access to its functionality until the update is fully downloaded and installed.
+
+## Steps
+
+1. Run a static analysis tool such as @MASTG-TOOL-0110 on codebase for usages of the calls to the Play Core in-app update API, specifically `startUpdateFlowForResult`, that are configured with the integer value `1`.
+
+## Observation
+
+The output file shows usages of the application calls `startUpdateFlowForResult` with `AppUpdateOptions.newBuilder(1).build()`. This is the exact signature for triggering a mandatory, non-cancellable update flow.
```

Comment:
`The output should contain ...`


Check https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?tab=t.0#heading=h.h9gqgz4hdubj